src/components/MainNav/MainNav.js

```diff
@@ -86,6 +99,39 @@ class MainNav extends Component {
     this.context.router.history.push('/');
   }
 
+  subscribeToQueryChanges(moduleInfo) {
+    if (this.unsub) this.unsub();
```

Comment:
Since you're subscribing to all state changes not those of any particular resource, rather than subbing/unsubbing repeatedly couldn't you just have `this.storeKeyToReflectInTheUrlIfAny` and update that as needed based on the `props.location` changes you're currently using to trigger this subscription?